src/functionaltests/java/com/ericsson/ei/subscriptions/trigger/SubscriptionTriggerSteps.java

```diff
@@ -1,44 +1,308 @@
+        } catch (IOException e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+
+        int port = SocketUtils.findAvailableTcpPort();
+        LOGGER.debug("Setting REST port to " + port);
+        restServer = startClientAndServer(port);
+        
+        // Set up endpoints
+        mockClient = new MockServerClient(BASE_URL, port);
+        mockClient.when(request().withMethod("POST").withPath(REST_ENDPOINT))
+                .respond(response().withStatusCode(201));
+        mockClient.when(request().withMethod("POST").withPath(REST_ENDPOINT_AUTH))
+                .respond(response().withStatusCode(201));
+        mockClient.when(request().withMethod("POST").withPath(REST_ENDPOINT_PARAMS))
+                .respond(response().withStatusCode(201));
+        mockClient.when(request().withMethod("POST").withPath(REST_ENDPOINT_AUTH_PARAMS))
+                .respond(response().withStatusCode(201));
     }
 
-    @Given("^Subscription mail trigger is created$")
-    public void subscription_mail_trigger_is_created() throws Throwable {
-        LOGGER.debug("Creating subscription mail trigger");
+    @After("@SubscriptionTriggerScenario")
+    public void afterScenario() {
+        LOGGER.debug("Stopping SMTP and REST Mock Servers");
+        smtpServer.stop();
+        try {
+            mockClient.close();
+        } catch (IOException e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+        restServer.stop();
     }
 
-    @Given("^Subscription rest authenticated trigger is created$")
-    public void subscription_rest_authenticated_trigger_is_created() throws Throwable {
-        LOGGER.debug("Subscription rest authenticated trigger is created");
+    @Given("^The REST API \"([^\"]*)\" is up and running$")
+    public void the_REST_API_is_up_and_running(String endPoint) {
+        RequestBuilder requestBuilder = MockMvcRequestBuilders.get(endPoint).accept(MediaType.APPLICATION_JSON);
+        try {
+            result = mockMvc.perform(requestBuilder).andReturn();
+            LOGGER.debug("Response code from mocked REST API: " + String.valueOf(result.getResponse().getStatus()));
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+        assertEquals(HttpStatus.OK.value(), result.getResponse().getStatus());
     }
 
-    @When("^I send events$")
-    public void send_events() throws Throwable {
-        LOGGER.debug("Sending events");
+    @Given("^Subscriptions is setup using REST API \"([^\"]*)\"$")
+    public void subscriptions_is_setup_using_REST_API(String endPoint) {
+        String readFileToString = "";
+        try {
+            readFileToString = FileUtils.readFileToString(new File(SUBSCRIPTION_WITH_JSON_PATH), "UTF-8");
+        } catch (IOException e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+        readFileToString = stringReplaceText(readFileToString);
+        
+        ArrayList<String> subscriptions = new ArrayList<>();
```

Comment:
why still ArrayList??? everywhere you use List but only here ArrayList, we must use interface realization 
List< String > subscriptions = new ArrayList<>();

Reply:
And what about move functionality for process events to another class, we have a lot of functional tests where we use rabbitmq i think it will be very useful to have this functionality in FunctionalTestBase. When you can just override methods

Reply:
Code has been updated.